dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleConnectorScope.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.oracle;
+
+import com.google.edwmigration.dumper.application.dumper.connector.LogsConnector;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.OracleLogsDumpFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.OracleMetadataDumpFormat;
+
+enum OracleConnectorScope {
```

Comment:
Optional: I don't have a strong opinion on this, but as already discussed offline, I would follow the existing convention of implementing connectors and avoid any changes in the `OracleMetadataConnector` and `OracleLogsConnector` classes in this PR. For the stats one, it should be enough to implement `Connector` with the `String getDefaultFileName(boolean isAssessment);` method and add new format as a `String` constant in the `OracleStatsConnector` class.

Later, a similar code as in this enum can be introduced as a refactoring across all the connectors.

Reply:
I prefer to use this enum for the Oracle connectors. I think it is an improvement compared to implementing the `LogsConnector` and `MetadataConnector` interfaces.

Implementing `Connector` for stats needs to be done through `AbstractOracleConnector`, because methods that we need are encapsulated in various classes of this hierarchy. And since we are making changes to the abstract oracle connector, we're able to use these changes in other Oracle connectors without much effort.

I agree that it would be better to have consistency across all connectors, but I would prefer to make use of this change in code that is currently being changed the most and propagate it to the old connectors later on.